stages/org.osbuild.kernel-cmdline.bls-append

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python3
+"""
+Add kernel command line parameters to a BLS [1] config either in
+the tree or in a mount.
+
+[1] https://freedesktop.org/wiki/Specifications/BootLoaderSpec/
+"""
+
+
+import glob
+import sys
+from urllib.parse import urlparse
+
+import osbuild.api
+
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "required": ["kernel_opts"],
+  "properties": {
+    "kernel_opts": {
+      "description": "Additional kernel command line options",
+      "type": "array",
+      "items": {
+        "description": "A single kernel command line option",
+        "type": "string"
+      }
+    },
+    "bootpath": {
+      "type": "string",
+      "description": "The mounted location of the boot filesystem tree where the BLS entries will be under ./loader/entries/*.conf",
+      "pattern": "^(mount|tree):\/\/\/",
+      "examples": ["tree:///boot", "mount:///", "mount:///boot"],
+      "default": "tree:///boot"
+    }
+  }
+},
+"devices": {
+  "type": "object",
+  "additionalProperties": true
+},
+"mounts": {
+  "type": "array"
+}
+"""
+
+
+def main(paths, tree, options):
+    kopts = options.get("kernel_opts", [])
+    bootpath = options.get("bootpath", "tree:///boot")
+
+    url = urlparse(bootpath)
+    scheme = url.scheme
+    if scheme == "tree":
+        root = tree
+    elif scheme == "mount":
+        root = paths["mounts"]
+    else:
+        raise ValueError(f"Unsupported scheme '{scheme}'")
```

Comment:
I was looking at the first commit again now and was going to say we should do exactly this.  The caller should be explicit about whether to use the tree or the mount.

Does the `root = paths["mounts"]` work or does it need to get the `"path"` from the sub-object too?
(see https://github.com/osbuild/osbuild/blob/d52738d70c5841613387e6dcd60e1a2f18c56d27/stages/org.osbuild.copy#L90-L99)

Reply:
> Does the `root = paths["mounts"]` work or does it need to get the `"path"` from the sub-object too?
> Check out the `parse_mount()` function from the copy stage:

I'm not sure. I think I copied the `root = paths["mounts"]` from somewhere else and this seems to work, though I'm not sure if I'm just getting lucky or not.

Reply:
ok yeah here is where I copied it from:

https://github.com/osbuild/osbuild/blob/cf00c5b21482d572709e9a5353676d028ce587de/stages/org.osbuild.zipl.inst#L90